File: AspNetMvcUnleashed/0672329980_source/Chapter09/CS/MvcApplication1/MvcApplication1/Constraints/AuthenticatedConstraint.cs
using System.Web;
using System.Web.Routing;

namespace MvcApplication1.Constraints
{
    public class AuthenticatedConstraint : IRouteConstraint
    {

        public bool Match
            (
                HttpContextBase httpContext, 
                Route route, 
                string parameterName, 
                RouteValueDictionary values, 
                RouteDirection routeDirection
            )
        {
            return httpContext.Request.IsAuthenticated;
        }

    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter09/VB/MvcApplication1/MvcApplication1/Constraints/AuthenticatedConstraint.vb
Imports System.Web
Imports System.Web.Routing

Public Class AuthenticatedConstraint
    Implements IRouteConstraint

    Public Function Match _
    ( _
        ByVal httpContext As HttpContextBase, _
        ByVal route As Route, _
        ByVal parameterName As String, _
        ByVal values As RouteValueDictionary, _
        ByVal routeDirection As RouteDirection _
    ) As Boolean Implements IRouteConstraint.Match
        Return HttpContext.Request.IsAuthenticated
    End Function

End Class
